### ISSUE TITLE
Unraid docker ffmpeg nvenc transcoding not working

### ISSUE THREAD

**Issue by Skwisgaarz** [vexorian/dizquetv#431]
Been talking with vexorian on reddit but thought it would be beneficial to post it here too. so it can be flagged moving forward.

I have my Nvidia GPU transcoding working in plex with ich777's nvidia driver plug in, also have dizque set up right as it works with cpu transcoding (but buffers way too much to be useable, hence wanting to get the gpu to work here), but when I try to launch a stream in plex using h264_nvenc ffmpeg transcoding it gives a 'Could not tune channel. Please check your tuner or antenna.' error in plex. 

In the attached logs I get a 'Cannot load libcuda.so.1' error when it tries to initiate nvenc transcoding so apparently even with the nvidia plug in working correct the dizque docker won't communicate with the gpu.

I'm running a 1.5.1 nvidia test repository vexorian sorted for me with apparently the fix for this exact issue in the windows .exe ported in to a docker but the issue in the logs is still hapenning.

![Docker run](https://user-images.githubusercontent.com/115912817/196018726-47ea138b-ca71-4adf-b2a6-58a4f027712c.png)

[ffmpeg nvenc error logs.txt](https://github.com/vexorian/dizquetv/files/9793626/ffmpeg.nvenc.error.logs.txt)

**Comment by Skwisgaarz** [comment]
Response from ich777 on the unraid nvidia plugin forum:

"As alturismo already pointed out, this is a container issue itself, is the CUDA runtime installed inside the container, not all but a few applications need the CUDA runtime installed in the container.

 

Of course there are other workarounds for this but non of them which you can easily do, for example install the Nvidia driver inside the container, but if you do so, it has to match the driver which is installed on the host and this is where issues for you, the user, start."

**Comment by ich777** [comment]
@Skwisgaarz no, please read it again, I've updated the comment in the Unraid forums because I've read your issue over here after I wrote that.

You even don't pass through the Nvidia GPU to the Docker container, please read the second post on the Unraid Forums for the Nvidia Driver. You have to pass through the GPU to the container too, otherwise it can't work.
[Click](https://forums.unraid.net/topic/98978-plugin-nvidia-driver/?do=findComment&comment=913251)

**Comment by Skwisgaarz** [comment]
Ah legend, thanks for that, didn't see your update sorry.

**Comment by ich777** [comment]
Please close this issue if adding the variables to the container solves your problem.

**Comment by Skwisgaarz** [comment]
Sorry for the delay. I made the changes in the linked post. Added extra parameter '--runtime=nvidia', variable 'NVIDIA_VISIBLE_DEVICES' with my UUID, variable 'NVIDIA_DRIVER_CAPABILITIES' with the value 'all'. Nvenc transcoding definitely enabled in Dizque. Same error in plex when I start the stream. Dizque logs attached. 
[Logs post variable.txt](https://github.com/vexorian/dizquetv/files/9794179/Logs.post.variable.txt)
